A/abseil_cpp/build_tarballs.jl

```diff
@@ -3,21 +3,41 @@
 using BinaryBuilder, Pkg
 
 name = "abseil_cpp"
-version = v"20240116.2"
+version = v"20230125.0"
 
 # Collection of sources required to complete build
 sources = [
-    GitSource("https://github.com/abseil/abseil-cpp", "d7aaad83b488fd62bd51c81ecf16cd938532cc0a"),
-    DirectorySource("bundled"),
+    GitSource("https://github.com/abseil/abseil-cpp", "78be63686ba732b25052be15f8d6dee891c05749"),
+    DirectorySource("./bundled"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/abseil-cpp
 
-# Avoid problems with `-march`, `-ffast-math` etc.
-sed -i -e 's!set(CMAKE_C_COMPILER.*!set(CMAKE_C_COMPILER '${WORKSPACE}/srcdir/files/ccsafe')!' ${CMAKE_TARGET_TOOLCHAIN}
-sed -i -e 's!set(CMAKE_CXX_COMPILER.*!set(CMAKE_CXX_COMPILER '${WORKSPACE}/srcdir/files/c++safe')!' ${CMAKE_TARGET_TOOLCHAIN}
+# ABSL_RANDOM_HWAES_*_FLAGS are used to set ABSL_RANDOM_RANDEN_COPTS,
+# which is used when compiling absl/random/internal/randen_hwaes.cc
+
+# Output ABSL_RANDOM_RANDEN_COPTS during configure
+atomic_patch -p1 ../patches/cmake-copts.patch
+
+# Allow setting "-march=armv8-a+crypto"
+find $(dirname $(which $CC)) -type f \
+    | xargs grep --files-with-matches 'BinaryBuilder: Cannot force an architecture via -march' \
+    | xargs -n1 sed -i -E 's/^if \[\[ (" \$\{ARGS\[@\]\} " == \*"-march="\*) \]\]; then/if [[\n    \1\n    \&\& ! " \${ARGS[@]} " == *"-march=armv8-a+crypto"*\n]]; then/'
```

Comment:
I'm a bit confused: aren't you patching the CMake file not to set `-march=armv8-a+crypto`?

Reply:
No, I’m just patching it for **x86_64**-apple-darwin - which is just to remedy some slightly opaque cmake configuration mess from upstream which causes the ARM64 flag to be applied for x86_64-apple-darwin.

Reply:
… and the X64 flags to be applied for aarch64-apple-darwin 🔀

Reply:
That sounds like an upstream bug?

Reply:
20230125 is past the two year due date, so I would not expect patches upstream to be handled.
https://abseil.io/about/releases

Reply:
Newer versions don't have this problem?

Reply:
I don’t know yet - my plan is to push a range of PRs to cover all abseil versions used by protobuf - this is the first:
* https://protobuf.dev/reference/cpp/abseil/
* https://github.com/protocolbuffers/protobuf/commits/v22.0/third_party

Reply:
It definitely looks like they figured out something was bad around July 2024: https://github.com/abseil/abseil-cpp/pull/1710